lib/graphiti/adapters/graphiti_api.rb

```diff
@@ -0,0 +1,89 @@
+module Graphiti
+  module Adapters
+    class GraphitiAPI < ::Graphiti::Adapters::Null
+      def base_scope(model)
+        {}
+      end
+
+      def resolve(scope)
+        url = build_url(scope)
+        response = resource.make_request(url)
+        json = JSON.parse(response.body)
+
+        if json['errors']
+          handle_remote_error(url, json)
+        else
+          models = json['data'].map { |d| build_entity(json, d) }
+          Util::RemoteSerializer.for(resource.class.serializer, models)
+          models
+        end
+      end
+
+      private
+
+      def handle_remote_error(url, json)
+        errors = json['errors'].map do |error|
+          if raw = error['meta'].try(:[], '__raw_error__')
+            { message: raw['message'], backtrace: raw['backtrace'] }
+          else
+            { message: "#{error['title']} - #{error['detail']}" }
+          end
+        end.compact
+        raise Errors::Remote.new(url, errors)
+      end
+
+      def build_url(scope)
+        url    = resource.remote
```

Comment:
Thinking about a more robust `UserResource < UsersApiResource` subclassing setup for multiple resources from the same API, I'd like to be able to set either `remote` to be very explicit or something like `remote_base_url` on a base class, which would allow each child resource to either infer or explicitly set its resource name to append to build `remote`.

Reply:
Added `remote_base_url`, which is *optional*. `remote` can be a path or a full url depending on your use of this.